examples/hitl/rearrange_v2/app_state_base.py

```diff
@@ -51,13 +52,23 @@ def record_state(self):
 
     def _status_message(self, message: str) -> None:
         """Send a message to all users."""
+        ui_manager = self._app_service.ui_manager
+        with ui_manager.update_canvas("top_left", Mask.ALL) as ctx:
+            ctx.label(
+                uid="status",
+                text=message,
+                font_size=24,
+                horizontal_alignment=HorizontalAlignment.LEFT,
+            )
+        # Server-side message.
+        # TODO: Handle from UI manager.
```

Comment:
There is currently no local fallback.

This is a planned change - the UI system should provide a minimal local implementation.